mutation_partition.cc

```diff
@@ -2109,13 +2109,13 @@ stop_iteration query::result_memory_accounter::check_local_limit() const {
     } else {
         if (_total_used_memory > _maximum_result_size.hard_limit) {
             throw std::runtime_error(fmt::format(
-                    "Memory usage of unpaged query exceeds hard limit of {} (configured via max_memory_for_unlimited_query_hard_limit)",
-                    _maximum_result_size.hard_limit));
+                    "Memory usage of unpaged query {} exceeds hard limit of {} (configured via max_memory_for_unlimited_query_hard_limit)",
+                    _query_id, _maximum_result_size.hard_limit));
```

Comment:
Unpaged queries will have no  query uuid set. And I don't see any changes on the coordinator side, adding logs for failed queries that can be correlated with the replica side log messages.

Reply:
> Unpaged queries will have no query uuid set.

oh.  any deep reason for that (if there is, it obviously invalidates the whole approach here)?

> adding logs for failed queries

I assumed getting query ids in the coordinator log is just a matter of playing with verbosity, guess I was wrong here too...

Reply:
> oh. any deep reason for that (if there is, it obviously invalidates the whole approach here)?

Query uuid is set for stateful queries: queries that save their states between pages. This whole mechanism is bypassed for unpaged queries.

> I assumed getting query ids in the coordinator log is just a matter of playing with verbosity, guess I was wrong here too...

AFAIK only the query pager logs the query uuid when on debug/trace level. So this also doesn't work for unpaged queries.
Also, in general relying on debug/trace level logging to uncover the reason for failed queries doesn't work well. When you discover the error in the log, it is too late to turn on debug/trace level logging, unless you can reproduce -- but if you can reproduce you already know roughly where the problem is.

Reply:
Even worse, `query_uuid` effectively doubles as a boolean flag for "is this query paged".  Welp, so much for this PR.

Reply:
We can make the uuid always defined, for both unpaged and paged queries. The query id's double role as a flag for stateful queries was a pre-caution when this feature was developed -- we wanted a simple kill-switch in case there is some major bug with it. It has been years without any major issues with this, so maybe we can drop this secondary role and have all queries have an id.